### DISCUSSION TITLE
form error filed when an exception throwed in onSubmit function

### DISCUSSION THREAD

**Discussion by mh-alahdadian** [react-hook-form/react-hook-form#2530] [Ideas]
Hi
there are some forms which might get error from server in submit phase 
for example in login submit username and password might be mismatched and this error is not related to any  field and it is relate to form itself 
I think there should be a mechanism that when submit function throws handleSubmit method catch it and give it as a form-error in formMethods

**Comment by bluebill1049** [comment]
That's right. as today, you will have to use `setError` API until we figure out a good solution for serverside errors.

**Reply by bluebill1049** [reply]
you can vote this feature request as well: https://github.com/react-hook-form/react-hook-form/issues/1614
